viewing_party/party.py

```diff
@@ -1,23 +1,177 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie_dict = {}
+    if title == None or genre == None or rating == None:
+        return None
+    else:
+        movie_dict["title"] = title
+        movie_dict["genre"] = genre
+        movie_dict["rating"] = rating
+        return movie_dict
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    avg_rating = 0
+    for i in range(len(user_data["watched"])):
+        avg_rating += user_data["watched"][i]["rating"]
+    if len(user_data["watched"]) == 0:
+        return avg_rating
+    avg_rating /= len(user_data["watched"])
+    return avg_rating
+
+#second swing at get_watched_avg_rating. Which one is better?
+# def get_watched_avg_rating(user_data):
+#     user_watched = user_data["watched"]
+#     avg_rating = 0
+    
+#     for movie in user_watched:
+#         avg_rating += movie["rating"]
+#         print(avg_rating)
+#     if len(user_watched) == 0:
+#         return 0.0
+#     else:
+#         avg_rating /= len(user_watched)
+#         return avg_rating
+
+def get_most_watched_genre(user_data):
+    genre_dict = {}
+    i = 0
+    if len(user_data["watched"]) == 0:
+        return None
+    for key, value in user_data["watched"][i].items():
+        if key == "genre":
+            if value in genre_dict:
+                genre_dict[value] += 1
+            else:
+                genre_dict[value] = 1
+        i += 1
+    max_value = max(genre_dict.values())
+    for key, value in genre_dict.items():
+        if value == max_value:
+            return key
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+#returns unique movies that user has watched, but friends have not
+def get_unique_watched(user_data):
+    unique_watched_list = []
+
+    friend_watched_set = set(get_friends_watched(user_data))
+    user_watched_set = set(get_users_watched(user_data))
+
+    unique_set = user_watched_set - friend_watched_set
+
+    for movie in unique_set:
+        for i in range(len(user_data["watched"])):
+            if user_data["watched"][i]['title'] == movie:
+                unique_watched_list.append(user_data["watched"][i])   

```

Comment:
This for loop style works, but it is more "pythonic" to use the "for element in list" style of loop when possible:
```suggestion
    for movie in unique_set:
        for user_movie in user_data["watched"]:
            if user_movie['title'] == movie:
                unique_watched_list.append(user_movie)   

```